sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/WebJobsServiceBusTestBase.cs

```diff
@@ -217,6 +223,18 @@ internal async Task WriteTopicMessage(string message, string sessionId = null)
             }
             await sender.SendMessageAsync(messageObj);
         }
+
+        internal static async Task AssertQueueEmptyAsync(string queueName = default)
+        {
+            // TODO - see if this can be part of Host.StopAsync validation (currently it doesn't apply to all tests and would increase
+            // test durations unnecessarily)
+            var client = new ServiceBusClient(ServiceBusTestEnvironment.Instance.ServiceBusConnectionString);
+            var receiver = client.CreateReceiver(queueName ?? _firstQueueScope.QueueName);
+
+            // Poll for longer than lock duration specified when creating the queues (15s) to ensure message
+            // is really gone rather than just locked.
+            Assert.IsNull(await receiver.ReceiveMessageAsync(TimeSpan.FromSeconds(20)));
```

Comment:
![https://i.pinimg.com/originals/8d/b3/2a/8db32a5dd518fd17287dc4ff11890ef8.gif](https://i.pinimg.com/originals/8d/b3/2a/8db32a5dd518fd17287dc4ff11890ef8.gif)

Reply:
Can we use QueueRuntimeProperties for this?

Reply:
Hmm, I'm not sure if it will be as up to date as we would need. But I can look into it.

Reply:
Doesn't seem to be reliable.

Reply:
I can add a smaller delay though and use it.